### ISSUE TITLE
Add Way to Exclude Calls from VerifyNoOtherCalls

### ISSUE THREAD

**Issue by rarcher** [devlooped/moq#1490]
When verifying calls on mocks, it often happens that there is some trivial operation (like a property read) that you really don't care about, and the executing code is allowed to invoke any number of times.  However, because of the way `VerifyNoOtherCalls` works, there's no way to mark things as ignored - you _have_ to explicitly specify each individual call, which leads to unit tests either being too tightly coupled to the main code base, or using hackish work-arounds (like doing `Times.AtMost(1000)`).  There ought to be a way to tell Moq that such-and-such a call just isn't relevant.

[![Back this issue](https://raw.githubusercontent.com/devlooped/SponsorLink/main/docs/assets/img/separator.png "Back this issue")](https://www.devlooped.com/SponsorLink/github/issues/?s=devlooped)
[![Back this issue](https://img.shields.io/badge/backed-%240-EA4AAA?logo=githubsponsors "Back this issue")](https://www.devlooped.com/SponsorLink/github/issues/?s=devlooped)

**Comment by rarcher** [comment]
As side note, the most obvious way to implement this would be to let `Times.AtLeast` accept a `0` value as a parameter.

**Comment by kzu** [comment]
Wouldn't turn your mock into a loose mock and only verify what you're interested in work better in this case?

**Comment by rarcher** [comment]
For situations where there aren't many methods you need to verify, sure.  But the use case for this is when you have a LOT of functions, and you want to verify that most of them are never touched, with the possible exception of a few accessors.  In that scenario, it's much more convenient to handle everything in bulk, using `VerifyNoOtherCalls`.

**Comment by kzu** [comment]
If most should not be called, then a strict mock with no setups for them seems like a better approach?

**Comment by rarcher** [comment]
Hmmmm.  Fair point.  Maybe this is more of a documentation issue then?  It is not obvious that strict mocks are the way to handle this case, for someone with limited familiarity with the library.

**Comment by kzu** [comment]
Yeah, perhaps. I guess this is something that is typically covered by the multiple blogs/stackoverflow/online curses you can find by now on using Moq itself. But I'll keep that in mind for vNext.

Feel free to close this issue if you feel the strict mock will suffice.

Thanks!
